examples/tree_2d_dgsem/elixir_shallowwater_conical_island.jl

```diff
@@ -0,0 +1,105 @@
+
+using OrdinaryDiffEq
+using Trixi
+using Plots
+###############################################################################
+# semidiscretization of the shallow water equations
+
+equations = ShallowWaterEquations2D(gravity_constant=9.81, H0=1.4)
+
+cfl = 0.1                                    
+
+function initial_condition_conical_island(x, t, equations::ShallowWaterEquations2D)
+  # Set the background values
+  
+  v1 = 0.0
+  v2 = 0.0
+
+  x1, x2 = x
+  b = max(0.1, - sqrt(16*x1^2+16*x2^2) + 1)
+  
+  # use a logistic function to tranfer water height value smoothly
+  L  = equations.H0    # maximum of function
+  x0 = 0.3   # center point of function
+  k  = -25.0 # sharpness of transfer
+  
+  H = max(b, L/(1.0 + exp(-k*(sqrt(x1^2+x2^2) - x0))))
+
+  H = max(H, b + equations.threshold_limiter)
+  return prim2cons(SVector(H, v1, v2, b), equations)
+end
+
+initial_condition = initial_condition_conical_island
+# Works with Dirichlet boundary conditions as well
+#boundary_conditions = BoundaryConditionDirichlet(initial_condition)
```

Comment:
Ok, so you think I should use Dirichlet as a default case here? The interesting fact about this elixir (using periodic boundary) is that there is the nice wetting/drying and even running up the hill due to the periodic boundary. Using Dirichlet sets the boundaries nearly to 0 and simulates something like an outflow. Maybe, I could just create two elixirs?
You are right, in the most cases a different boundary did not work right now unfortunately...

Reply:
I do not think you need two elixir files. Here I would take Dirichlet to demonstrate that this is possible to produce meaningful results because periodic BCs are used elsewhere.

Reply:
I found out the following: Up to now, the `alpha_smooth` is only active for a periodic boundary on `StructuredMesh`. At least, it gave me an error message stating that. As I implemented this elixir both in `TreeMesh` and `StructuredMesh`, the periodic boundary (and therefore the nice wetting and drying) can be seen there.